workerapi/worker.thrift

```diff
@@ -0,0 +1,38 @@
+enum Status {
+  UNKNOWN = 0
+  READY = 1
+  PENDING = 2
+  FAILURE = 3
```

Comment:
We should assume until proven otherwise that the exit code is always non-zero on failure (I thought it could be zero). If that's the case then, sure, COMPLETE plus an exit code payload is probably sufficient.